### PR TITLE
ENH: Add multi-backend tests for bnb

### PR DESCRIPTION
**Pull request by younesbelkada**
bitsandbytes has now a new branch `multi-backend-refactor`: https://github.com/TimDettmers/bitsandbytes/tree/multi-backend-refactor to conduct experiments and push forward the efforts for cross-platform support for bnb

This PR extends the current existing testing suite to make sure we also conduct the same tests on bitsandbytes compiled on that branch 

cc @Titus-von-Koeller @pacman100 @BenjaminBossan

### CODE REVIEW

File: docker/peft-gpu-bnb-multi-source/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    python3 -m pip install -U --no-cache-dir \
+    librosa \
+    "soundfile>=0.12.1" \
+    scipy \
+    git+https://github.com/huggingface/transformers \
+    git+https://github.com/huggingface/accelerate \
+    peft[test]@git+https://github.com/huggingface/peft \
+    optimum \
+    auto-gptq && \
+    git clone https://github.com/TimDettmers/bitsandbytes && cd bitsandbytes && git checkout multi-backend-refactor && \
```
Comment (younesbelkada) [inline docker/peft-gpu-bnb-multi-source/Dockerfile:59]: This file is a copy of: https://github.com/huggingface/peft/blob/main/docker/peft-gpu-bnb-source/Dockerfile except for this line where we checkout to multi-backend-refactor branch

**Review by pacman100 — APPROVED** [review]
Thank you @younesbelkada, LGTM, good to merge once the CI is green!

File: docker/peft-gpu-bnb-multi-source/Dockerfile (hunk 1 above)
Comment (BenjaminBossan) [reply docker/peft-gpu-bnb-multi-source/Dockerfile:59]: Could this be added as a comment? Or perhaps, we should even add a README.md to `peft/docker/` that mentions something like: When making changes to x/Dockerfile, also change y/Dockerfile.

**Review by BenjaminBossan — APPROVED** [review]
Thanks, LGTM. I added a small comment, but no ~~docker~~ blocker.

**Pull request merged**